extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCryptoKeyStore.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2018 The Data-Portability Project Authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dataportabilityproject.cloud.google;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.cloudkms.v1.CloudKMS;
+import com.google.api.services.cloudkms.v1.CloudKMSScopes;
+import com.google.api.services.cloudkms.v1.model.DecryptRequest;
+import com.google.api.services.cloudkms.v1.model.DecryptResponse;
+import com.google.inject.Inject;
+import java.io.IOException;
+import org.dataportabilityproject.cloud.google.GoogleCloudModule.ProjectId;
+import org.dataportabilityproject.spi.cloud.storage.CryptoKeyStore;
+
+/**
+ * Crypto key management using Google's Cloud KMS.
+ */
+final class GoogleCryptoKeyManagementSystem implements CryptoKeyStore {
+  // Key for encrypting app secrets.
+  private static final String SECRETS_CRYPTO_KEY_FMT_STRING = "projects/%s/locations/global/"
+      + "keyRings/portability_secrets/cryptoKeys/portability_secrets_key";
+
+  private CloudKMS cloudKMS;
+  private String secretsCryptoKey;
+
+  @Inject
+  GoogleCryptoKeyManagementSystem(@ProjectId String projectId) throws GoogleCredentialException {
+    HttpTransport transport = new NetHttpTransport();
```

Comment:
Make transport injectable or add a TODO to do so

Reply:
Done, did same with Json factory